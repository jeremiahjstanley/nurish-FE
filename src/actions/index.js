export const addFormulas = (formulas) => ({
  type: 'ADD_FORMULAS',
  formulas
})

export const formulaFetchDataSuccess = (formulas) => ({
  type: 'FORMULA_FETCH_DATA_SUCCESS',
  formulas
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (bool) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
})