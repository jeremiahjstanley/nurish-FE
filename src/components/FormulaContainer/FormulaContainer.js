import React from 'react'
import { withRouter, NavLink } from 'react-router-dom';



const FormulaContainer = () => {
	return (
		<div>
			<NavLink to='/'>back</NavLink>
			<NavLink to='/search'>search</NavLink>
			<NavLink to='/calculate'>calculate</NavLink>
			<p>I'm a formula container</p>
		</div>
	)
}

export default withRouter(FormulaContainer);