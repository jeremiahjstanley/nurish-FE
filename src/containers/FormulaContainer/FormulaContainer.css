.formula-container {
  width: 50%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap; }

div .formula-container {
  width: 85%;
  animation: grow 1s; }

@keyframes grow {
  0% {
    width: 25%; }
  100% {
    width: 85%; } }
