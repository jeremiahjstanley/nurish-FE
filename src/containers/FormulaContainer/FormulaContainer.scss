.formula-cards {
	grid-column-gap: 3%;
	margin: 2rem auto;
	width: 95%;
  display: grid;
  grid-row-gap: 20px;
  grid-template-columns: 32% 32% 32%;
  justify-content: space-evenly;
}

div {
	& .formula-container {
		animation: grow-container 1s;
		height: fit-content;
		margin: 0 auto 5rem;
		max-width: 70rem;
		width: 85%;
		@keyframes grow-container {
  
	  	0% {
	  	  width: 25%;
	  	}
	  
	  	100% {
	    	width: 85%;
	  	}
		}
	}	
}
