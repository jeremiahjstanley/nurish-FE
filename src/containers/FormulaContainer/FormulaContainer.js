import React, { Component } from 'react';
import { withRouter, NavLink } from 'react-router-dom';
import { formulas } from '../../data-helper/mockFormula.js';
import { FormulaCard } from '../../components/FormulaCard/FormulaCard';
import './FormulaContainer.css';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';


export class FormulaContainer extends Component{


  FormulaCards = () => {
    console.log(formulas)

    return formulas.map(formula => {
      return (
        <FormulaCard
          key={formula.id}
          {...formula}
        />
        )
    })
  }

  render() {
  	return (
  		<div className='form-container formula-container'>
        <div className='link-container'>
          <NavLink to='/'>back</NavLink>
  			  <NavLink to='/search'>search</NavLink>
          <NavLink to='/filter'>filter</NavLink>
        </div>
        {this.FormulaCards()}
  		</div>
  	)
  }
}

export const mapStateToProps = (state) => {
  formulas: state.formulas
}

export default withRouter(connect(mapStateToProps, null)(FormulaContainer));
