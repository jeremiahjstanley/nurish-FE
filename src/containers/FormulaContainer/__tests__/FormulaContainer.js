import React from 'react';
import { shallow } from 'enzyme';
import {FormulaContainer, mapStateToProps, mapDispatchToProps} from '../FormulaContainer';
import * as action from '../../../actions';

describe('FormulaContainer', () => {
  let wrapper
  let mockFilterResultCard
  let mockFormulaCard
  let mockAddFilterResults
  let mockFilterResults
  let mockAnimateLogo

  beforeEach(() => {
    mockFormulaCard = jest.fn();
    mockFilterResultCard = jest.fn();
    mockAddFilterResults = jest.fn();
    mockFilterResults = [{title: 'boost'}, { title: 'smoothie'}];
    mockAnimateLogo = jest.fn()

    wrapper = shallow(<FormulaContainer
      formulaCard={mockFormulaCard}
      filterResultCard={mockFilterResultCard}
      filterResults={mockFilterResults}
      mockAddFilterResults={mockAddFilterResults}
      animateLogo={mockAnimateLogo}
      />);
  });

  it('should match the snapshot when there are filtered results', () => {
    let mockFormulas = [{title: 'boost'}, { title: 'smoothie'}];

    wrapper = shallow(<FormulaContainer
      formulaCard={mockFormulaCard}
      filterResultCard={mockFilterResultCard}
      filterResults={[]}
      mockAddFilterResults={mockAddFilterResults}
      formulas={mockFormulas}
      animateLogo={mockAnimateLogo}
      />);

    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot when there are no filtered results', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call animateLogo when mounted', () => {
    wrapper.instance()

    expect(mockAnimateLogo).toHaveBeenCalledWith('collapse-logo')
  });

  it('should call animateLogo when unmounted', () => {
    wrapper.instance().componentWillUnmount()

    expect(mockAnimateLogo).toHaveBeenCalledWith('expand-logo')
  });

  it('should return a filterResults array', () => {
    const mockState = {
      filterResults: mockFilterResults,
      addFilterResults: mockAddFilterResults
    };

    const expectedProps = {
      filterResults: mockFilterResults
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expectedProps);
  });
});