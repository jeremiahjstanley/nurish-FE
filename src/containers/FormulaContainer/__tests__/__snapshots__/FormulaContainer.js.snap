// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormulaContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormulaContainer
    animateLogo={[Function]}
    filterResultCard={[Function]}
    filterResults={
        Array [
            Object {
              "title": "boost",
            },
            Object {
              "title": "smoothie",
            },
          ]
    }
    formulaCard={[Function]}
    mockAddFilterResults={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="link-container"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
          >
                    back
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/search"
          >
                    search
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/filter"
          >
                    filter
          </NavLink>
</div>,
        <div
          className="formula-cards"
>
          <Unknown
                    title="boost"
          />
          <Unknown
                    title="smoothie"
          />
</div>,
      ],
      "className": "form-container formula-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
>
              back
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/search"
>
              search
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/filter"
>
              filter
</NavLink>,
          ],
          "className": "link-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "back",
              "to": "/",
            },
            "ref": null,
            "rendered": "back",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "search",
              "to": "/search",
            },
            "ref": null,
            "rendered": "search",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "filter",
              "to": "/filter",
            },
            "ref": null,
            "rendered": "filter",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              title="boost"
/>,
            <Unknown
              title="smoothie"
/>,
          ],
          "className": "formula-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "boost",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "smoothie",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="link-container"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/"
            >
                        back
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/search"
            >
                        search
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/filter"
            >
                        filter
            </NavLink>
</div>,
          <div
            className="formula-cards"
>
            <Unknown
                        title="boost"
            />
            <Unknown
                        title="smoothie"
            />
</div>,
        ],
        "className": "form-container formula-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
>
                back
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/search"
>
                search
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/filter"
>
                filter
</NavLink>,
            ],
            "className": "link-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "back",
                "to": "/",
              },
              "ref": null,
              "rendered": "back",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "search",
                "to": "/search",
              },
              "ref": null,
              "rendered": "search",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "filter",
                "to": "/filter",
              },
              "ref": null,
              "rendered": "filter",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                title="boost"
/>,
              <Unknown
                title="smoothie"
/>,
            ],
            "className": "formula-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "boost",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "smoothie",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
